### PR TITLE
Improve description of work with dynamic data in D3 challenge

### LINKED ISSUE #14668: [Beta] Suggested Description Changes—Work with Dynamic Data in D3 Challenge
Challenge [work-with-dynamic-data-in-d3](http://beta.freecodecamp.com/en/challenges/data-visualization-with-d3/work-with-dynamic-data-in-d3) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30</code>.

Part of the trigger for raising this issue is the incorrect use of "couple", instead of "couple of", in the D3 challenge in question (id: 587d7fa7367417b2b2512bc5).  Another instance can be found in the challenge **Use the d3.max and d3.min Functions to Find Minimum and Maximum Values in a Dataset** (id: 587d7fac367417b2b2512bdc).

If I'm not mistaken, the use of simple past when describing a fact, such as in the sentence "The last couple **of** challenges cover~~ed~~ the tools...", is also incorrect.

I think the description in the title challenge could use a rework—there are many ideas being explained but, in my opiniong, the current text is potentially confusing, particularly to non-native speakers.  For example, the sentence *"since the data from the dataset array is attached to each element, the callback function has access to it"* appears to be misleading because, if I'm not mistaken, each ```h2``` element only has access to an item in ```dataset``` that the element is mapped to.

I am more than happy to suggest a reworked description, but am not entirely sure how to go about it; not least because these changes can be subjective and I'm not entirely confident that I will be able to explain those concepts clearly with my lack of coding experience.

Many apologies in advance for any inconvenience caused—I have't had to report any issues or used GitHub beyond basic version control up until now.  :(

### PR DESCRIPTION
**Pull request by dakshshah96**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #14668 

#### Description

Implemented the suggested changes to description of the "Work with Dynamic Data in D3" challenge.

### CODE REVIEW

**Comment by QuincyLarson** [comment]
@HKuz do you have any feedback on this copy change?

**Comment by HKuz** [comment]
I had left my comments over on the original issue. I'm on mobile, so hard to compare, but looks like this PR makes the recommended changes. 👍

File: seed/challenges/04-data-visualization/data-visualization-with-d3.json [hunk 1]
```diff
@@ -168,12 +168,14 @@
...
-        "The last challenge created a new <code>h2</code> for each item in the data array, but it displayed the same text (\"New Title\") for each heading. Fortunately, there is a way to access and display the actual data with callback functions.",
-        "The <code>text()</code> method can take a string or a callback function as an argument. Since the data from the <code>dataset</code> array is attached to each element, the callback function has access to it. The parameter used in the callback function (<code>d</code> in the example below) is for the individual data-point itself. This callback function would set the text in the selection to the data value:",
+        "The last two challenges cover the basics of displaying data dynamically with D3 using the <code>data()</code> and <code>enter()</code> methods. These methods take a data set and, together with the the <code>append()</code> method, create a new DOM element for each entry in the data set.",
```
Comment (dhcodes) [inline seed/challenges/04-data-visualization/data-visualization-with-d3.json]: The word "the" is used twice in this line. One of them needs to be removed.

File: seed/challenges/04-data-visualization/data-visualization-with-d3.json [hunk 2]
```diff
@@ -168,12 +168,14 @@
...
+        "In the previous challenge, you created a new <code>h2</code> element for each item in the <code>dataset</code> array, but they all contained the same text, \"New Title\". This is because you have not made use of the data that is bound to each of the <code>h2</code> elements.",
+        "The D3 <code>text()</code> method can take a string or a callback function as an argument:",
         "<code>selection.text((d) => d)</code>",
+        "In the example above, the parameter <code>d</code> refers to a single entry in the dataset that a selection is bound to.",
+        "Using the current example as context, the first <code>h2</code> element is bound to 12, the second <code>h2</code> element is bound to 31, the third <code>h2</code> element is bound to 22, and so on. When you provide a callback function with <code>d</code> as a parameter to the <code>text()</code> method, it means that <code>d</code> will take a value of 12 for the first <code>h2</code> element, 31 for the second <code>h2</code> element, 22 for the third <code>h2</code> element, and so on.",
```
Comment (dhcodes) [inline seed/challenges/04-data-visualization/data-visualization-with-d3.json]: The end of this line seems to be repetitive of the beginning.

**Review by dhcodes — CHANGES_REQUESTED** [review]

**Comment by QuincyLarson** [comment]
@dakshshah96 It looks like this is almost ready to merge. Can you take a look at the changes requested by @dhcodes?

**Comment by dakshshah96** [comment]
@QuincyLarson Done!

**Comment by BerkeleyTrue** [comment]
@dakshshah96 Merged! Happy Coding!

**Comment by QuincyLarson** [comment]
@dakshshah96 Thanks again for your contribution!

**Pull request merged**